### PR TITLE
Replace "for of" by classic for loop (#759)

### PR DESCRIPTION
**Pull request by ctjhoa**
#759

### CODE REVIEW

**Comment by ctjhoa** [comment]
@bantic not sure why CI failed as I do not modify templates

**Comment by mixonic** [comment]
@ctjhoa I'd like to keep this using `for`, as that avoids a function allocation and execution.

File: addon/-private/collapse-tree.js [hunk 1]
```diff
@@ -275,7 +277,9 @@ export const TableRowMeta = EmberObject.extend({
 function reduceSelectedRows(selection, groupingCounts, rowMetaCache) {
   let reducedGroupingCounts = new Map();
 
-  for (let [group, count] of groupingCounts.entries()) {
+  let groupingCountsEntries = Array.from(groupingCounts.entries());
```
Comment (mixonic) [inline addon/-private/collapse-tree.js]: It is unfortunate to allocate an array for each of the maps.

But furthermore `entries` is a method on `Map`, and though IE11 has a `Map` implementation it doesn't have support for `entries`. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/entries#Browser_compatibility

So there is still a need for a polyfil in IE11.

**Comment by ctjhoa** [comment]
Ok I'm giving up on this one. I'll open another PR to document `includePolyfill`

**Pull request closed without merging**